benchmarks/benchmark_ocl.py

```diff
@@ -0,0 +1,161 @@
+import tempfile
+from pathlib import Path
+
+from kedro.config import OmegaConfigLoader
+
+base_catalog = {
+    "dataset_1": {
+        "type": "pandas.CSVDataset",
+        "filepath": "data1.csv"
+    },
+    "dataset_2": {
+        "type": "pandas.CSVDataset",
+        "filepath": "data2.csv"
+    },
+    "dataset_3": {
+        "type": "pandas.CSVDataset",
+        "filepath": "data3.csv"
+    },
+    "dataset_4": {
+        "type": "pandas.CSVDataset",
+        "filepath": "data4.csv",
+        "versioned": True,
+    },
+}
+local_catalog = {
+    "dataset_4" : {
+        "filepath": "data4_local.csv",
+        "type": "pandas.CSVDataset",
+    },
+    "dataset_5" : {
+        "filepath": "data5_local.csv",
+        "type": "pandas.CSVDataset",
+    },
+}
+base_params = {
+    "param_1": "value_1",
+    "param_2": "value_2",
+    "param_3": "value_3",
+    "param_4": "value_4",
+}
```

Comment:
Can we have some extreme test to test OmegaConfigLoader is still performing ok when dealing with a large amount of config? i.e. 10000+ configuration. I have seen big projects that has 20+ pipelines and take half a min just to load config (not with OmegaConfigLoader).

Reply:
Updated to larger configs